core/src/main/java/jenkins/security/ConfidentialStore.java

```diff
@@ -121,15 +121,21 @@ void clear() {
         }
 
         @Override
-        protected void store(ConfidentialKey key, byte[] payload) throws IOException {
-            LOGGER.fine(() -> "storing " + key.getId() + " " + Util.getDigestOf(Util.toHexString(payload)));
-            data.put(key.getId(), payload);
+        protected void store(ConfidentialKey key, byte[] payload) throws IllegalStateException {
+                byte[] payloadDigest = Util.getSHA256DigestOf(payload);
+                String payloadDigestHexString = (payloadDigest != null) ? Util.toHexString(payloadDigest) : null;
+                //called only from tests
+                LOGGER.fine("storing " + key.getId() + " " + payloadDigestHexString);
```

Comment:
unconditional string concatination should be avoided in logging statements.  Use parameterized logging or a supplier.